pkg/new-ui/v1beta1/frontend/src/app/models/experiment.k8s.model.ts

```diff
@@ -0,0 +1,164 @@
+import { K8sObject } from 'kubeflow';
+import { V1Container } from '@kubernetes/client-node';
+
+/*
+ * K8s object definitions
+ */
+export interface ExperimentK8s extends K8sObject {
```

Comment:
I think we should add Kubernetes metadata to `ExperimentK8s` object interface also.
Then we can properly show the Experiment YAML here with metadata: https://github.com/kubeflow/katib/blob/bc676bdca76ce64b807d7cb0e562875519d81bba/pkg/new-ui/v1beta1/frontend/src/app/pages/experiment-details/yaml/experiment-yaml.component.ts#L14-L16

Reply:
It already does. The `ExperimentK8s` interface extends the `K8sObject` interface, which contains the `metadata` definition.

https://github.com/kubeflow/katib/blob/bc676bdca76ce64b807d7cb0e562875519d81bba/pkg/new-ui/v1beta1/frontend/src/app/models/experiment.k8s.model.ts#L7

https://github.com/kubeflow/kubeflow/blob/0c4a738625f52092f66ec462e800ddd2837e0f6d/components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/utils/kubernetes.model.ts#L5

Reply:
Got it, thank you!
Is it possible to include also `TypeMeta` in the Experiment, like in the APIs: https://github.com/kubeflow/katib/blob/master/pkg/apis/controller/experiments/v1beta1/experiment_types.go#L279.
Then we don't need to manually add TypeMeta (like you did here: https://github.com/kubeflow/katib/blob/bc676bdca76ce64b807d7cb0e562875519d81bba/pkg/new-ui/v1beta1/frontend/src/app/pages/experiment-creation/experiment-creation.component.ts#L82-L83)  and use values from the models.

Currently, in the YAML we also don't have this TypeMeta:
![Screenshot 2021-02-19 at 23 08 50](https://user-images.githubusercontent.com/31112157/108571175-8186ea00-7307-11eb-9c6d-4fa35a65fc09.png)

Reply:
Just a sec, in the above message you mentioned the YALM contents from two different pages, the form for creating new Experiments and the YAML tab in the Experiment's details.

I agree that we should add the TypeMeta in the `K8sObject` definition in the common library, but the two components mentioned above are not directly affected from this because:
1. The form's JS will need to construct the Experiment CR that will be shown/submitted. So it will need to manually set the TypeMeta one way or the other
2. In the YAML tab the reason we don't see the TypeMeta [apiVersion and kind] is because the backend does not return this information

Reply:
> 1. The form's JS will need to construct the Experiment CR that will be shown/submitted. So it will need to manually set the TypeMeta one way or the other.

Got it. What do you think about importing `TypeMeta` from the models ? Then we can control Experiment version in the models and not in the experiment-creation page:
https://github.com/kubeflow/katib/blob/master/pkg/apis/controller/experiments/v1beta1/experiment_types.go#L279

> 2. In the YAML tab the reason we don't see the TypeMeta [apiVersion and kind] is because the backend does not return this information

You are right. I will check the backend later to see why it doesn't return `TypeMeta`

Reply:
Pushed another commit that
1. defines the `apiVersion` and `kind` values to use in a common place, the models file
2. changed the experiment-creation page to use the value from the models file for the TypeMeta